### PR TITLE
[API] Support streaming in get_logs

### PR DESCRIPTION
**Pull request by TomerShor**
In this PR we change `get_logs` to always return a response stream along with the run_id.
The response stream can either be iterated over, or just use its response regularly. 
This is backwards compatible as the http client can handle a stream response without iterations.

### CODE REVIEW

File: mlrun/db/httpdb.py [hunk 1]
```diff
@@ -405,6 +414,15 @@ def store_log(self, uid, project="", body=None, append=False):
...
         self.api_call("POST", path, error, params, body)
 
+    def stream_log(self, uid: str, project: str = ""):
+        """Stream a log from the API
+        :param uid: Log unique ID
+        :param project: Project name for which this log belongs
+        """
+        path = self._path_of("log-stream", project, uid)
+        error = f"stream log {project}/{uid}"
+        self.api_call("GET", path, error, stream=True, to_stdout=True)
```
Comment (Tankilevitch) [inline mlrun/db/httpdb.py:424]: seems like its not being used and we also don't have `log-stream` endpoint.

File: mlrun/db/httpdb.py [hunk 2]
```diff
@@ -405,6 +414,15 @@ def store_log(self, uid, project="", body=None, append=False):
         error = f"store log {project}/{uid}"
         self.api_call("POST", path, error, params, body)
 
+    def stream_log(self, uid: str, project: str = ""):
+        """Stream a log from the API
+        :param uid: Log unique ID
+        :param project: Project name for which this log belongs
+        """
```
Comment (Tankilevitch) [inline mlrun/db/httpdb.py:421]: I would add a TODO and a jira ticket for us to change the client to use the `stream_log` and get rid from our current implementation

File: mlrun/api/utils/clients/log_collector.py [hunk 3]
```diff
@@ -104,20 +105,21 @@ async def get_logs(
         while True:
             try:
                 response_stream = self._call_stream("GetLogs", request)
-                logs = b""
                 async for chunk in response_stream:
                     if not chunk.success:
+                        print(chunk.error)
```
Comment (Tankilevitch) [inline mlrun/api/utils/clients/log_collector.py:110]: redundant

File: mlrun/api/utils/clients/log_collector.py [hunk 4]
```diff
@@ -104,20 +105,21 @@ async def get_logs(
...
                             )
                         if verbose:
                             logger.warning(msg, error=chunk.error)
-                    logs += chunk.logs
-                return logs
+                    yield chunk.logs
+                return
             except Exception as exc:
                 try_count += 1
+                print(f"Failed to get logs, retrying ({try_count})")
```
Comment (Tankilevitch) [inline mlrun/api/utils/clients/log_collector.py:122]: should use logger

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
looks good, minor changes needed

File: mlrun/db/httpdb.py (hunk 2 above)
Comment (TomerShor) [reply mlrun/db/httpdb.py:421]: Well, I think I'll remove it altogether. `get_log` uses stream anyways

**Pull request merged**